### PR TITLE
fix: gpuoperator patched image with fixed CVE

### PR DESCRIPTION
**Pull request by SandhyaRavi2403**
**What problem does this PR solve?**:
Patch k8s-device-plugin:v0.15.0-ubi8-d2iq.0 and gpu-operator-validator image with fixed CVE's

workflow link
https://github.com/mesosphere/dkp-container-images/actions/runs/9962429325
https://github.com/mesosphere/dkp-container-images/actions/runs/9902973870
`sandhya.ravi@GT9X7CVF5F kommander-applications % trivy image ghcr.io/mesosphere/dkp-container-images/nvcr.io/nvidia/cloud-native/gpu-operator-validator:v24.3.0-d2iq.0  --vuln-type os --ignore-unfixed | grep Total 2024-07-16T16:32:13+05:30	INFO	Vulnerability scanning is enabled 2024-07-16T16:32:13+05:30	INFO	Secret scanning is enabled 2024-07-16T16:32:13+05:30	INFO	If your scanning is slow, please try '--scanners vuln' to disable secret scanning 2024-07-16T16:32:13+05:30	INFO	Please see also https://aquasecurity.github.io/trivy/v0.53/docs/scanner/secret#recommendation for faster secret detection 2024-07-16T16:32:21+05:30	INFO	Detected OS	family="redhat" version="8.9" 2024-07-16T16:32:21+05:30	INFO	[redhat] Detecting RHEL/CentOS vulnerabilities...	os_version="8" pkg_num=195 2024-07-16T16:32:22+05:30	WARN	Using severities from other vendors for some vulnerabilities. Read https://aquasecurity.github.io/trivy/v0.53/docs/scanner/vulnerability#severity-selection for details. Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)`

**Which issue(s) does this PR fix?**:

https://jira.nutanix.com/browse/NCN-101394

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Checklist**

- [ ] If the PR adds a version bump, ensure there is no breaking change in Licensing model (or NA).
- [ ] If a chart is changed or app configuration is significantly changed, the chart version is correctly incremented (so that apps are not automatically upgraded from a previous version of DKP).

### CODE REVIEW

File: services/nvidia-gpu-operator/24.3.1/defaults/cm.yaml [hunk 1]
```diff
@@ -33,4 +33,7 @@ data:
       enabled: true
       version: 3.3.5-3.4.1-ubuntu22.04
     validator:
-      version: v24.3.0
+      version: v24.3.0-d2iq.0
```
Comment (mhrabovcin) [inline services/nvidia-gpu-operator/24.3.1/defaults/cm.yaml:36]: Don't we need to override repository in these similarly to operator?

Comment (SandhyaRavi2403) [reply services/nvidia-gpu-operator/24.3.1/defaults/cm.yaml:36]: I added it for operator as the mapping wasn't happening. but for validator I think not an issue as the image is also a part of **services/nvidia-gpu-operator/24.3.1/helmrelease/extra-images.txt**.

Comment (mhrabovcin) [reply services/nvidia-gpu-operator/24.3.1/defaults/cm.yaml:36]: The `extra-images` file defines additional images that cannot be gathered from helm chart itself. Its usually a case when an operator (installed by a helm chart) uses this value to launch additional deployments. So its not directly visible. I think that in this case, if we only provide new version without the repository, the GPU operator will fail when launched.

Here you need to check the GPU operator helm chart and see if repositories for these images can be provided to the operator.

Comment (SandhyaRavi2403) [reply services/nvidia-gpu-operator/24.3.1/defaults/cm.yaml:36]: I referred to the below one 
https://github.com/NVIDIA/gpu-operator/blob/bfefa4d700783a3e1e4db07bf0e30840818abe96/deployments/gpu-operator/values.yaml#L67

Comment (mhrabovcin) [reply services/nvidia-gpu-operator/24.3.1/defaults/cm.yaml:36]: https://github.com/NVIDIA/gpu-operator/blob/bfefa4d700783a3e1e4db07bf0e30840818abe96/deployments/gpu-operator/values.yaml#L52-L54 - validator image is searched in the `repository: nvcr.io/nvidia/cloud-native`. If only the `version: v24.3.0-d2iq.0` is changed the config would still try to pull data from `nvcr.io/nvidia/cloud-native`, where the `v24.3.0-d2iq.0` doesn't exist. This config needs to be updated add

```
      repository: ghcr.io/mesosphere/dkp-container-images/nvcr.io/nvidia/cloud-native
      version: v24.3.0-d2iq.0
```

The same applies on the `devicePlugin` section - https://github.com/NVIDIA/gpu-operator/blob/bfefa4d700783a3e1e4db07bf0e30840818abe96/deployments/gpu-operator/values.yaml#L243-L245

Comment (SandhyaRavi2403) [reply services/nvidia-gpu-operator/24.3.1/defaults/cm.yaml:38]: updated

**Review by mhrabovcin — APPROVED** [review]
🚀

**Pull request merged**